src/brogue/Config.c

```diff
@@ -0,0 +1,255 @@
+#include "../cjson/cJSON.h"
+#include "Rogue.h"
+#include "GlobalsBase.h"
+
+typedef struct configParams {
+    short playbackDelayPerTurn;
+    short gameVariant;
+    short graphicsMode;
+    boolean displayStealthRangeMode;
+    boolean trueColorMode;
+    boolean wizard;
+    boolean easyMode;
+} configParams;
+
+typedef enum {
+    INT_TYPE,
+    BOOLEAN_TYPE,
+    ENUM_STRING // a json string that needs to be mapped to an Enum
+} fieldType;
+
+typedef struct {
+    int min;
+    int max;
+} intRange;
+
+typedef union {
+    const char** enumMappings;
+    intRange intRange;
+} fieldValidator;
+
+typedef struct {
+    const char* name;
+    void* paramPointer; // a pointer to a field of configParams struct
+    fieldType type;
+    fieldValidator validator;
+} configField;
+
+static const char* JSON_FILENAME = "brogue_config.json";
+
+// maps json strings to the gameVariant Enum
+static const char* variantMappings[] = {"brogue", "rapid", NULL};
+
+// maps json strings to the graphicsModes Enum
+static const char* graphicsModeMappings[] = {"text", "tiles", "hybrid", NULL};
+
+static intRange playbackDelayRange = {MIN_PLAYBACK_DELAY, MAX_PLAYBACK_DELAY};
+
+static configParams createDefaultConfig() {
+    configParams config;
+
+    config.playbackDelayPerTurn = DEFAULT_PLAYBACK_DELAY;
+    config.gameVariant = VARIANT_BROGUE;
+    config.graphicsMode = TEXT_GRAPHICS;
+    config.displayStealthRangeMode = false;
+    config.trueColorMode = false;
+    config.easyMode = false;
+    config.wizard = false;
+
+    return config;
+}
+
+static char* loadConfigFile() {
+    FILE* jsonFile = fopen(JSON_FILENAME, "r");
+
+    if (!jsonFile) {
+        return NULL;
+    }
+
+    fseek(jsonFile, 0, SEEK_END);
+    long fileSize = ftell(jsonFile);
+    fseek(jsonFile, 0, SEEK_SET);
+
+    char* buffer = (char*)malloc(fileSize + 1);
+
+    if (!buffer) {
+        fclose(jsonFile);
+        return NULL;
+    }
+
+    fread(buffer, 1, fileSize, jsonFile);
```

Comment:
nit: Should probably check the number of bytes written by `fread`, and return an error if it's less than `fileSize`